File: src/main/java/com/sica/backend/dto/ApiResponse.java
package com.sica.backend.dto;

public record ApiResponse<T>(
    Boolean success,
    String message,
    T data
) {
    public ApiResponse(T data) {
        this(true, null, data);
    }

    public ApiResponse(Boolean success, String message) {
        this(success, message, null);
    }
}

File: src/main/kotlin/com/sica/backend/dto/ApiResponse.kt
package com.sica.backend.dto

data class ApiResponse<T>(
    val success: Boolean = true,
    val message: String? = null,
    val data: T? = null
)

data class ErrorResponse(
    val success: Boolean = false,
    val error: String,
    val message: String? = null,
    val timestamp: Long = System.currentTimeMillis()
)
